app/product_commands.go

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package app
+
+import (
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/mattermost/mattermost-server/v6/app/request"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/shared/mlog"
+)
+
+// Code inspired on plugin_commands.go.
+//
+// Key differences/points:
+// - There's no need of health checks or unregisterProductCommands on products, they are compiled and assumed as active server side
+// - HooksForProduct still returns a plugin.Hooks struct, it might make sense to improve the name/package
+// - Plugin code had a check for a plugin crash after a command was executed, that has been omitted for products
```

Comment:
Could we just embed these into `plugin_commands.go`? I mean adding some conditonals to branching for product and plugins. The reason why I'm asking is once we start to add tests, it shouldn't be a duplicated work and result with inconsistencies.